x-pack/platform/plugins/shared/fleet/server/services/outputs/helpers.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { agentPolicyService } from '../agent_policy';
+import { appContextService } from '../app_context';
+
+import { LEGACY_AGENT_POLICY_SAVED_OBJECT_TYPE } from '../../constants';
+import { SO_SEARCH_LIMIT } from '../../../common/constants';
+
+// Returns agentless policies that may need their data output ID updated
+// If outputId is provided, return agentless policies that use that output in addition
+// to policies that don't have an output set
+export async function findAgentlessPolicies(outputId?: string) {
+  const internalSoClientWithoutSpaceExtension =
+    appContextService.getInternalUserSOClientWithoutSpaceExtension();
+
+  const agentlessPolicies = await agentPolicyService.list(internalSoClientWithoutSpaceExtension, {
```

Comment:
shouldn't we query policies from all spaces like this? https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fleet/server/services/agent_policy.ts#L927-L938

Reply:
I checked with Nicolas and when we query with SO client directly, it should be using `namespaces: ['*']`, but when using our services it should be with `spaceId: '*'`

here I am using `agentPolicyService` with a non-scoped soClient so the query looks correct to me

Reply:
👍  the property is named namespaces in the saved object, but we expose it and the rest of kibana (and documentation) expose it to the end user as space id, and to add to the confusion we have another concept of namespace in fleet policies